tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1833,8 +1875,15 @@ static bool CombineBasicTypes(
       case BPROP_BITS0:  *pOutKind = AR_BASIC_LITERAL_INT; break;
       case BPROP_BITS8:  *pOutKind = AR_BASIC_INT8;        break;
       case BPROP_BITS12: *pOutKind = AR_BASIC_MIN12INT;    break;
-      case BPROP_BITS16: (uEitherFlags & BPROP_MIN_PRECISION) ?
-        *pOutKind = AR_BASIC_MIN16INT : *pOutKind = AR_BASIC_INT16; break;
+      case BPROP_BITS16:
+        if (IS_BPROP_16BIT_SINT_NO_MIN_PRECISION(uLeftProps) ||
+            IS_BPROP_16BIT_SINT_NO_MIN_PRECISION(uRightProps)) {
```

Comment:
Similar comment to above, both are guaranteed to be signed integers, so why not `BPROP_MIN_PRECISION & uBothFlags` with reversed then/else?